src/Microsoft.Restier.Core/Query/QueryExpressionContext.cs

```diff
@@ -215,33 +214,35 @@ private QueryModelReference ComputeModelReference()
                                 var typeOfT = sourceType.GetGenericArguments()[0];
                                 if (parameter.Type == typeOfT)
                                 {
-                                    modelReference = new CollectionElementReference(modelReference);
-                                    break;
+                                    var collectionType = modelReference.Type as IEdmCollectionType;
+                                    if (collectionType != null)
+                                    {
+                                        modelReference = new QueryModelReference(modelReference.EntitySet, collectionType.ElementType.Definition);
```

Comment:
avoid deep nest and break this line. It's too long. StyleCop will catch this line.